### PR TITLE
Update doc for CA1506 to mention exclusion of generated types

### LINKED ISSUE #4988: CA1506 - Generated types are no longer counting as coupled
With the release of fxcop 3.X, the analyzer will stop counting generated types as coupled.

See https://github.com/dotnet/roslyn-analyzers/pull/3315

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ed6ab2e8-1f95-1fa5-4dfc-0d34d745436a
* Version Independent ID: d56217c8-0c2c-000f-2fd0-01c240fde547
* Content: [CA1506: Avoid excessive class coupling - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1506?view=vs-2019)
* Content Source: [docs/code-quality/ca1506.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1506.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4988

This new feature is not yet publicly released. Please sync-up with @mavasani for when to merge this PR.

### CODE REVIEW

**Comment by PRMerger14** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1506.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ ms.workload:
 
 ## Cause
 
-A type or method is coupled with many other types.
+A type or method is coupled with many other types excluding generated types.
```
Comment (mavasani) [inline docs/code-quality/ca1506.md:29]:
```suggestion
A type or method is coupled with many other types. Compiler generated types are excluded from this metric.
```

**Comment by ktoliver** [comment]
Hi @Evangelink - Could you make the revisions that @mavasani requests? Thanks.

**Comment by PRMerger14** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Mikejo5000** [comment]
#sign-off

**Pull request merged**